### PR TITLE
Don't persist documents that we already have

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
Port of https://github.com/firebase/firebase-android-sdk/pull/720

### CODE REVIEW

File: Firestore/Example/Tests/Util/FSTHelpers.mm [hunk 1]
```diff
@@ -304,14 +304,25 @@ MaybeDocumentMap FSTTestDocUpdates(const std::vector<MaybeDocument> &docs) {
 
 RemoteEvent FSTTestAddedRemoteEvent(const MaybeDocument &doc,
                                     const std::vector<TargetId> &addedToTargets) {
-  HARD_ASSERT(!doc.is_document() || !Document(doc).has_local_mutations(),
-              "Docs from remote updates shouldn't have local changes.");
-  DocumentWatchChange change{addedToTargets, {}, doc.key(), doc};
+  const std::vector<MaybeDocument> docs{doc};
```
Comment (var-const) [inline Firestore/Example/Tests/Util/FSTHelpers.mm]: Nit: we don't use "shallow" `const` on local variables and function arguments (that is, `const` references and pointers to `const` are okay, but non-pointer non-reference variables shouldn't be `const`). While I personally like `const`, this is inconsistent with the rest of the codebase.

File: Firestore/Source/Local/FSTLocalStore.mm [hunk 2]
```diff
@@ -258,36 +258,25 @@ - (void)setLastStreamToken:(nullable NSData *)streamToken {
...
   return self.persistence.run("Apply remote event", [&] {
     // TODO(gsoltis): move the sequence number into the reference delegate.
     ListenSequenceNumber sequenceNumber = self.persistence.currentSequenceNumber;
 
-    DocumentKeySet authoritativeUpdates;
     for (const auto &entry : remoteEvent.target_changes()) {
       TargetId targetID = entry.first;
       const TargetChange &change = entry.second;
 
       // Do not ref/unref unassigned targetIDs - it may lead to leaks.
```
Comment (var-const) [inline Firestore/Source/Local/FSTLocalStore.mm]: I think this comment was removed on other platforms?

File: Firestore/Example/Tests/Util/FSTHelpers.mm (hunk 1 above)
Comment (schmidt-sebastian) [reply Firestore/Example/Tests/Util/FSTHelpers.mm]: That makes sense. Updated.

File: Firestore/Source/Local/FSTLocalStore.mm (hunk 2 above)
Comment (schmidt-sebastian) [reply Firestore/Source/Local/FSTLocalStore.mm]: Removed. Thanks for catching!

**Pull request merged**